examples/modal/app.R

```diff
@@ -0,0 +1,58 @@
+library(shiny)
+library(shiny.semantic)
+library(shinyjs)
+
+ui <- function() {
+  shinyUI(
+    semanticPage(
+      useShinyjs(),
+      title = "Modal example",
+      uiOutput("modalBind"),
+      uiOutput("modalAction"),
+      div(id = "modal-render-button", class = "ui button", "Im a button binded to a modal with output content"),
+      modal(
+        uiOutput("modalRenderContent"),
+        id = "render-example-modal",
+        header = "This is a modal example with output content",
+        class = "tiny",
+        target = "modal-render-button",
+        settings = list(c("transition", "fade"), c("closable", "false"))),
+      div(id = "modal-button", class = "ui button", "Im a button binded by the modal target parameter"),
+      actionButton("show", "I am an action button calling show_modal")
+    )
+  )
+}
+
+server <- shinyServer(function(input, output) {
+  observeEvent(input$show, {
+    show_modal('action-example-modal')
+    shinyjs::delay(5000, remove_modal('action-example-modal'))
+  })
+
+  output$modalBind <- renderUI({
+    modal(
+      "This is some example content",
+      id = "bind-example-modal",
+      header = "This is a modal example",
+      class = "tiny",
+      target = "modal-button",
+      settings = list(c("transition", "fade"), c("closable", "false")))
+  })
+
+  output$modalAction <- renderUI({
+    modal(
+      "I am another modal that is hidden by hide_modal after a few seconds",
+      id = "action-example-modal",
+      header = "This is a modal example",
+      footer = "",
+      class = "tiny",
+      settings = list(c("transition", "fade"), c("closable", "false")))
+
+  })
+  output$modalRenderContent <- renderUI({
+    div("This is some example content that is inside an output on the server")
+  })
+
+})
```

Comment:
Still, this example is far for the simplicity offered by standard shiny. @pedrocoutinhosilva, please take a look at an example:
```
shinyApp(
  ui = basicPage(
    actionButton("show", "Show modal dialog")
  ),
  server = function(input, output) {
    observeEvent(input$show, {
      showModal(modalDialog(
        title = "Important message",
        "This is an important message!",
        easyClose = TRUE
      ))
    })
  }
)
```

Reply:
shinyjs dependency as been removed, so it will no longer be required.

I also added a few more examples, but split into one example per file to avoid confusion.
Some of the best examples were also added directly to the documentation.

Reply:
Yeah, but still this what we have is quite different from the example above, don't you think? In your examples you keep defining modal in ui, whereas here everything is handled on the server side. I think that with what we have so far it should be easy to create such equivalent of `showModal`. Thanks.